test/test_abstraction.py

```diff
@@ -10,22 +10,23 @@ def test_h1_kwargs(self):
         c = HTTPConnection(
             'test', 443, secure=False, window_manager=True, enable_push=True,
             ssl_context=False, proxy_host=False, proxy_port=False,
-            other_kwarg=True
+            socket_timeout=None, other_kwarg=True
         )
 
         assert c._h1_kwargs == {
             'secure': False,
             'ssl_context': False,
             'proxy_host': False,
             'proxy_port': False,
+            'socket_timeout': None,
             'other_kwarg': True,
         }
 
     def test_h2_kwargs(self):
         c = HTTPConnection(
             'test', 443, secure=False, window_manager=True, enable_push=True,
             ssl_context=True, proxy_host=False, proxy_port=False,
-            other_kwarg=True
+            socket_timeout=None, other_kwarg=True
```

Comment:
Ideally we want to set this to a non-None value to confirm it's not getting defaulted. Let's try....`10`.